core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -543,6 +543,30 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging with Seria
       }
     }
 
+    if (contains("spark.cores.max")) {
+      val totalCores = getInt("spark.cores.max", -1)
+      if (totalCores <= 0) {
+        throw new IllegalArgumentException(s"spark.cores.max (was ${get("spark.cores.max")})" +
+          s" can only be a positive number")
+      }
+    }
+    if (contains("spark.executor.cores")) {
+      val executorCores = getInt("spark.executor.cores", -1)
+      if (executorCores <= 0) {
+        throw new IllegalArgumentException(s"spark.executor.cores " +
+          s"(was ${get("spark.executor.cores")}) can only be a positive number")
+      }
+    }
+    if (contains("spark.cores.max") && contains("spark.executor.cores")) {
+      val totalCores = getInt("spark.cores.max", 1)
```

Comment:
@srowen Users may set these configuration via SparkConf object in a programmatic way after arg checking for spark-submit. For example, an app code may be like this: "val conf = new SparkConf().set("spark.executor.cores","-1")" before SparkContext created, the checkings here can get the error and exit directly.